openpype/hosts/maya/plugins/publish/extract_playblast.py

```diff
@@ -138,28 +158,41 @@ def process(self, instance):
         override_viewport_options = (
             capture_presets['Viewport Options']['override_viewport_options']
         )
-        with lib.maintained_time():
-            filename = preset.get("filename", "%TEMP%")
-
-            # Force viewer to False in call to capture because we have our own
-            # viewer opening call to allow a signal to trigger between
-            # playblast and viewer
-            preset['viewer'] = False
-
-            # Update preset with current panel setting
-            # if override_viewport_options is turned off
-            if not override_viewport_options:
-                panel_preset = capture.parse_view(instance.data["panel"])
-                panel_preset.pop("camera")
-                preset.update(panel_preset)
-
-            self.log.info(
-                "Using preset:\n{}".format(
-                    json.dumps(preset, sort_keys=True, indent=4)
+
+        # Force viewer to False in call to capture because we have our own
+        # viewer opening call to allow a signal to trigger between
+        # playblast and viewer
+        preset['viewer'] = False
+
+        # Update preset with current panel setting
+        # if override_viewport_options is turned off
+        if not override_viewport_options:
+            panel_preset = capture.parse_view(instance.data["panel"])
+            panel_preset.pop("camera")
+            preset.update(panel_preset)
+
+        # Need to ensure Python 2 compatibility.
+        # TODO: Remove once dropping Python 2.
+        if getattr(contextlib, "nested", None):
+            # Python 3 compatibility.
+            with contextlib.nested(
+                lib.maintained_time(),
+                panel_camera(instance.data["panel"], preset["camera"])
+            ):
+                path = self._capture(
+                    preset, override_viewport_options, instance
+                )
+        else:
+            # Python 2 compatibility.
+            with contextlib.ExitStack() as stack:
+                stack.enter_context(lib.maintained_time())
+                stack.enter_context(
+                    panel_camera(instance.data["panel"], preset["camera"])
                 )
-            )
 
-            path = capture.capture(log=self.log, **preset)
+                path = self._capture(
+                    preset, override_viewport_options, instance
+                )
```

Comment:
Looking at the code the returned `path` isn't used other than debug logging down below. Should we move that debug log into the capture method and return nothing.

It does feel a bit weird that we're using `filename` instead of `path` that it returned from capture? No idea if that returns anything usable.

But currently `path` variable seems quite redundant to pass around.

Reply:
Yeah, lets get rid of that `path` variable.